processes/ingest/hathi_trust.py

```diff
@@ -85,37 +92,38 @@ def returnHathiDateFormat(strDate):
         else:
             return '%Y-%m-%d %H:%M:%S %z'
 
-    def importFromHathiFile(self, hathiURL):
+    def importFromHathiFile(self, hathi_url, start_date_time=None):
         try:
-            hathiResp = requests.get(hathiURL, stream=True, timeout=30)
-            hathiResp.raise_for_status()
+            hathi_resp = requests.get(hathi_url, stream=True, timeout=30)
+            hathi_resp.raise_for_status()
         except (ReadTimeout, HTTPError) as e:
-            logger.error('Unable to read hathifile url {}'.format(hathiURL))
+            logger.error('Unable to read hathifile url {}'.format(hathi_url))
             logger.debug(e)
             return None
 
-        with gzip.open(BytesIO(hathiResp.content), mode='rt') as hathiGzip:
-            hathiTSV = csv.reader(hathiGzip, delimiter='\t')
-            self.readHathiFile(hathiTSV)
+        with gzip.open(BytesIO(hathi_resp.content), mode='rt') as hathiGzip:
+            hathi_tsv = csv.reader(hathiGzip, delimiter='\t')
+            self.readHathiFile(hathi_tsv, start_date_time)
 
-    def readHathiFile(self, hathiTSV):
-        processed_record_count = 0
-
-        while True:
-            if self.ingest_limit and processed_record_count >= self.ingest_limit:
+    def readHathiFile(self, hathi_tsv, start_date_time=None):
+        for number_of_books_ingested, book in enumerate(hathi_tsv):
+            if self.ingest_limit and number_of_books_ingested > self.ingest_limit:
                 break
-
-            try:
-                row = next(hathiTSV)
-                logger.info('Reading row')
-            except csv.Error as e:
-                logger.warning('Unable to read TSV row')
-                logger.debug(e)
+            
+            if len(book) >= 2:
+                book_right = book[2]
+                if len(book) >= 15:
```

Comment:
if start_date_time is None should we continue if we aren't able to get the book_date_updated?

Reply:
The else continue conditional for the `book_date_updated` variable is unnecessary since we we should only check if the variable exists before we set the `hathi_date_modified` variable. I'll remove the nested else conditional and make a new if conditional for the `hathi_date_modified` variable like this:

```
if book_date_updated:
                hathi_date_modified = datetime.strptime(book_date_updated, '%Y-%m-%d %H:%M:%S').replace(tzinfo=None)
```